desc/coils.py

```diff
@@ -1474,6 +1572,34 @@ def compute(
             )
         ]
 
+    def compute_position(self, params=None, source_grid=None, **kwargs):
+        """Compute coil positions accounting for stellarator symmetry.
+
+        Parameters
+        ----------
+        params : dict or array-like of dict, optional
+            Parameters to pass to coils, either the same for all coils or one for each.
+        source_grid : Grid, int or None, optional
+            Grid used to discretize coils. If an integer, uses that many equally spaced
+            points. Should NOT include endpoint at 2pi.
+
+        Returns
+        -------
+        x : ndarray, shape(len(self),source_grid.num_nodes,3)
+            Coil positions, in [R,phi,Z] or [X,Y,Z] coordinates.
+
+        """
+        # TODO: catch error if coils have different default grids
```

Comment:
what is this error? we assume every sub coil is being computed with the same grid?

Reply:
We assume every coil in the coilset has the same compute grid (aka source grid). This happens when we stack together the big array of distances, which has shape (ncoils, num_nodes, 3), so if some coils have a grid with a different number of nodes then they won't concatenate together properly. 

This is only an issue for the `MixedCoilSet`, which can have different types of coils with different default grids. The objective `CoilsetMinDistance` takes care of this by making sure the same grid is used for every coil, but you can still run into problems if you call `MixedCoilSet.compute_position` directly.

Reply:
should we make this a private function then?